packages/labs/task/src/task.ts

```diff
@@ -115,21 +115,47 @@ export class Task<
   private _onError?: (error: unknown) => unknown;
   status: TaskStatus = TaskStatus.INITIAL;
 
+  /**
+   * Controls if they task will run when its arguments change. Defaults to true.
+   */
+  autoRun = true;
+
   /**
    * A Promise that resolve when the current task run is complete.
    *
    * If a new task run is started while a previous run is pending, the Promise
    * is kept and only resolved when the new run is completed.
    */
-  taskComplete!: Promise<R>;
+  get taskComplete(): Promise<R> {
+    // If a task exists, return the cached promise. This is true in the case
+    // where the user has called taskComplete in pending or completed state
+    // before and has not run a new task since.
+    if (this._taskComplete) {
```

Comment:
Optional, some 🏌️ :

```ts
return this._taskComplete ??= this.status === TaskStatus.PENDING
  ? new Promise((res, rej) => {
      this._resolveTaskComplete = res;
      this._rejectTaskComplete = rej;
    })
  : Promise.resolve(this._value!);
```

(except with your nice comments added back in)

Reply:
oh yikes, veto!

Reply:
Resolving with ACK as I think clarity is more important than the unbundled byte size here

Reply:
fwiw, I like the readability with the logical and assignment and ternary. The logical and assignment tells me that this is using the current value if available, or computing a new one. The ternary tells me that we're computing a single value based on a condition. To me that's more apparent than with the if/else block with multiple assignment statements form, since any other logic could be in the if/else blocks.

It also compresses better even after minification because there's only one reference to `this._taskComplete` instead of five. `this.` is something a minifier can't get rid of even if it changes the property name.